plugins/inputs/logfile/tailersrc_test.go

```diff
@@ -145,29 +145,30 @@ func TestTailerSrc(t *testing.T) {
 	}
 
 	// Removal of log file should stop tailersrc
-	if err := os.Remove(file.Name()); err != nil {
-		t.Errorf("failed to remove log file '%v': %v", file.Name(), err)
-	}
+	err = os.Remove(file.Name())
+	assert.NoError(t, err)
+	
+	time.Sleep(time.Second)
+	assert.Equal(t, 0, numUsedFds.GetCount())
+	
 	<-done
+	numUsedFds.Done()
```

Comment:
This means we never call `Done` if an exception occurs earlier. I'd prefer putting this call at the beginning in a defer